src/client/Microsoft.Identity.Client/Platforms/net45/NetDesktopDeviceAuthManager.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Permissions;
+using System.Threading.Tasks;
+#if DESKTOP
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Platforms.net45.Native;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.Identity.Client.Platforms.net45
+{
+    internal class NetDesktopDeviceAuthManager : IDeviceAuthManager
+    {
+        public bool CanHandleDeviceAuthChallenge
+        {
+            get { return true; }
+        }
+
+        public async Task<string> CreateDeviceAuthChallengeResponseAsync(IDictionary<string, string> challengeData)
+        {
+            string authHeaderTemplate = "PKeyAuth {0}, Context=\"{1}\", Version=\"{2}\"";
+
+            X509Certificate2 certificate = null;
+            try
+            {
+                certificate = FindCertificate(challengeData);
+            }
+            catch (MsalException ex)
+            {
+                if (ex.ErrorCode == MsalError.DeviceCertificateNotFound)
+                {
+                    return await Task.FromResult(string.Format(CultureInfo.InvariantCulture, @"PKeyAuth Context=""{0}"",Version=""{1}""", challengeData["Context"], challengeData["Version"])).ConfigureAwait(false);
+                }
+            }
+
+            DeviceAuthJWTResponse response = new DeviceAuthJWTResponse(challengeData["SubmitUrl"],
+                challengeData["nonce"], Convert.ToBase64String(certificate.GetRawCertData()));
+            CngKey key = NetDesktopCryptographyManager.GetCngPrivateKey(certificate);
+            byte[] sig = null;
+            using (RSACng rsa = new RSACng(key))
+            {
+                rsa.SignatureHashAlgorithm = CngAlgorithm.Sha256;
+                sig = rsa.SignData(response.GetResponseToSign().ToByteArray());
+            }
+
+            string signedJwt = string.Format(CultureInfo.InvariantCulture, "{0}.{1}", response.GetResponseToSign(),
+                Base64UrlHelpers.Encode(sig));
+            string authToken = string.Format(CultureInfo.InvariantCulture, " AuthToken=\"{0}\"", signedJwt);
+            Task<string> resultTask =
+                Task.Factory.StartNew(
+                    () =>
+                    {
+                        return string.Format(CultureInfo.InvariantCulture, authHeaderTemplate, authToken,
+                            challengeData["Context"],
+                            challengeData["Version"]);
+                    });
+
+            return await resultTask.ConfigureAwait(false);
```

Comment:
YOu don't need to do this trick with the factory, it's simpler to have just a 

` Task.FromResult(stringResult)`